File: Data Structures & Algorithms/transpose-matrix/submission-0.py
class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        n, m = len(matrix), len(matrix[0])
        if n == m:
            for i in range(n):
                for j in range(i+1):
                    temp = int(matrix[i][j])
                    matrix[i][j] = matrix[j][i]
                    matrix[j][i] = temp

            return matrix

        res = [[0 for _ in range(n)] for _ in range(m)]
        for i in range(m):
            for j in range(n):
                res[i][j] = matrix[j][i]

        return res

File: Data Structures & Algorithms/anagram-groups/submission-0.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {

        unordered_map<string, vector<string>> m;

        for(auto &x: strs){
            string temp = x;
            sort(temp.begin(), temp.end());
            if (m.find(temp) == m.end()){
                m[temp] = {x};
            }
            else{
                m[temp].push_back(x);
            }
        }
        vector<vector<string>> res;

        for(auto x: m){
            res.push_back(x.second);
        }
        return res;
    }
};
